includes/enrolment/class-sensei-course-enrolment.php

```diff
@@ -317,6 +325,46 @@ private function query_enrolment_providers( $user_id ) {
 		return $enrolment_results;
 	}
 
+	/**
+	 * Store the enrolment results in user meta.
+	 *
+	 * @param int                                      $user_id           User ID.
+	 * @param Sensei_Course_Enrolment_Provider_Results $enrolment_results Enrolment results object.
+	 */
+	private function store_enrolment_results( $user_id, Sensei_Course_Enrolment_Provider_Results $enrolment_results ) {
+		$results_meta_key   = $this->get_enrolment_results_meta_key();
+		$store_results      = true;
+		$had_existing_value = ! empty( get_user_meta( $user_id, $results_meta_key, true ) );
+
+		if (
+			! self::$store_negative_enrolment_results
+			&& ! $had_existing_value
+			&& ! $enrolment_results->is_enrolment_provided()
```

Comment:
We could write this like:
```
$store_results = self::$store_negative_enrolment_results || $had_existing_value || $enrolment_results..
```

Reply:
Nice! Changed in aee6f9cf6843907f5735a532d996e308f28dd36f